autometa/taxonomy/ncbi.py

```diff
@@ -170,114 +157,113 @@ def lineage(self, taxid, canonical=True):
-        df = pd.DataFrame(taxids_).transpose()
+                    ranked_taxids[taxid].update({rank: name})
+        df = pd.DataFrame(ranked_taxids).transpose()
         df.index.name = 'taxid'
         if fillna:
             df.fillna(value='unclassified', inplace=True)
         return df
 
     def rank(self, taxid):
-        """Short summary.
+        """
+        Takes a taxid and returns its rank
 
         Parameters
         ----------
         taxid : int
-            `taxid` to retrieve rank from `nodes`.
+            identifer for a taxon in the Taxonomy Database by NCBI
+            `taxid` to retrieve rank from `nodes` or `merged`
 
         Returns
         -------
         str
-            rank name
-
-        Raises
-        -------
-        ValueError
-            Provided `taxid` is not an integer
-
+            rank name if taxid is found in nodes.dmp or merged.dmp else 'unclassified'
         """
-        # If taxid is not found in nodes.dmp, will return 'unclassified'
-        try:
-            taxid = int(taxid)
-        except ValueError as err:
-            logger.error(
-                f'Taxid must be an integer! {taxid} type --> {type(taxid)}')
-            return None
-        return self.nodes.get(taxid, {'rank': 'unclassified'}).get('rank')
+        if not self.is_valid_taxid(taxid):
+            return False
+        # We have already checked that the taxid is present in either 'nodes.dmp' or 'merged.dmp'
+        # in the `is_valid_taxid() function` thus we don't need a default value
+        rank_name = self.nodes.get(taxid).get('rank')
+        if rank_name == None:
+            return self.merged.get(taxid).get('rank')
+        return rank_name
 
     def parent(self, taxid):
-        """Retrieve the parent taxid of provided taxid
+        """
+        Retrieve the parent taxid of provided taxid
 
         Parameters
         ----------
         taxid : int
-            Description of parameter `taxid`.
+            identifer for a taxon in the Taxonomy Database by NCBI
 
         Returns
         -------
         int
-            parent taxid
+            parent taxid if taxid is found in nodes.dmp or merged.dmp else 1
         """
-        # If taxid is not found in nodes.dmp, will return 1
-        try:
-            taxid = int(taxid)
-        except ValueError as err:
-            logger.error(
-                f'Taxid must be an integer! {taxid} type --> {type(taxid)}')
-            return None
-        return self.nodes.get(taxid, {'parent': 1}).get('parent')
+        if not self.is_valid_taxid(taxid):
+            return False
+        # We have already checked that the taxid is present in either 'nodes.dmp' or 'merged.dmp'
+        # in the `is_valid_taxid() function` thus we don't need a default value
+        parent_taxid = self.nodes.get(taxid).get('parent')
+        if parent_taxid == None:
+            return self.merged.get(taxid).get('parent')
+        return parent_taxid
```

Comment:
You have removed defaults from the `dict.get` method. However these provide functionality that would be useful here. What _should_ be done if a taxid is not found in the dict? Do you want to return a default value of `None` or something else? How about with `self.merged`? If the taxid is valid and up-to-date. Then it should not be in the dict of old values. 

Consider what you have removed `self.nodes.get(taxid, {'parent': 1}).get('parent')`.

We can break this down, but essentially we are providing defaults if the taxid is _invalid_. 

Just an example to consider when thinking about using the `dict.get` method.
```python
# Either
parent = self.nodes.get(taxid, {'parent': 1}).get('parent')
# or
parent = self.nodes.get(taxid, {'parent': None}).get('parent')
# Will depend on what the expected behavior *should* be. Consider the context of the method
parent = self.merged.get(taxid, {'parent': parent}).get('parent')
```